util/parsenum.h

```diff
@@ -29,6 +30,13 @@ _Pragma("clang diagnostic pop")
 #define PARSENUM_EPILOGUE /* NOTHING */
 #endif /* !__clang__ */
 
+/* Print a message before assert(). */
+#define ASSERT_MSG(x, s)						\
+	(								\
+		printf("ASSERT: " s "\n"),				\
```

Comment:
Bug #1: This ought to go to stderr, not stdout.
Bug #2: This should only get printed if x is zero.  Which it is in this case, of course... maybe drop that parameter and call this macro ASSERT_FAIL ?

While we're here, can you printf all of the assertion output and then call abort(), rather than printing a string ourselves and having assert() print more?